mlir/lib/Dialect/Arith/IR/ArithOps.cpp

```diff
@@ -1352,9 +1352,11 @@ OpFoldResult arith::TruncIOp::fold(FoldAdaptor adaptor) {
       setOperand(src);
       return getResult();
     }
+
     // trunci(zexti(a)) -> a
     // trunci(sexti(a)) -> a
-    return src;
+    if (srcType == dstType)
+      return src;
   }
```

Comment:
Should we land these fold fixes separately from the CSE change?

Reply:
Yes, these are part of #75887. (This commit depends on #75887.)